fdbserver/MoveKeys.actor.cpp

```diff
@@ -405,15 +588,48 @@ ACTOR Future<Void> cleanUpSingleShardDataMove(Database occ,
 				throw operation_cancelled();
 			}
 
-			std::vector<UID> src;
-			std::vector<UID> dest;
-			UID srcId, destId;
+			state std::vector<UID> src;
+			state std::vector<UID> dest;
+			state UID srcId;
+			state UID destId;
 			decodeKeyServersValue(UIDtoTagMap, currentShards[0].value, src, dest, srcId, destId);
 
 			if (dest.empty() || destId != anonymousShardId) {
 				return Void();
 			}
 
+			// Pre validate consistency of update of keyServers and serverKeys
+			if (SERVER_KNOBS->AUDIT_DATAMOVE_PRE_CHECK &&
+			    auditStorageFailedCount < SERVER_KNOBS->AUDIT_DATAMOVE_MAX_SUCCESSIVE_FAILED) {
+				state std::vector<Future<Void>> validationActors;
+				state std::unordered_map<UID, bool> validationResult;
+				for (auto& ssid : src) {
+					validationActors.push_back(
+					    store(validationResult[ssid],
+					          validateRangeAssignment(&tr, keys, ssid, "cleanUpSingleShardDataMove_precheck_src")));
+				}
+				for (auto& ssid : dest) {
+					validationActors.push_back(
+					    store(validationResult[ssid],
+					          validateRangeAssignment(&tr, keys, ssid, "cleanUpSingleShardDataMove_precheck_dest")));
+				}
+				wait(waitForAll(validationActors));
+				bool anyErrorDetected = false;
+				for (const auto& [ssid, consistent] : validationResult) {
+					if (!consistent) {
+						anyErrorDetected = true;
+						TraceEvent(SevError, "AuditStorageError")
```

Comment:
prefer `CorruptionInLocationMetadata` to `AuditStorageError`.

Reply:
Yes.